src/edu/csus/ecs/pc2/clics/API202306/CLICSContestState.java

```diff
@@ -0,0 +1,149 @@
+public class CLICSContestState {
+
+    @JsonProperty
+    private String started;
+
+    @JsonProperty
+    private String frozen;
+
+    @JsonProperty
+    private String ended;
+
+    @JsonProperty
+    private String thawed;
+
+    @JsonProperty
+    private String finalized;
+
+    @JsonProperty
+    private String end_of_updates;
+
+    /*
+     * These are non-standard PC2 specific properties to support pausing the contest
+     * 'paused' is the CONTEST elapsed time at which the contest was paused.  If set, it means the contest
+     * is currently paused. (HH:MM:SS.uuu)
+     *
+     * 'resumed' is the real (wall) time the contest was last resumed (or started, including initial contest start).
+     * You can tell if the contest had been paused at some point if resumed != started (ISO time)
+     */
+    @JsonProperty
+    private String paused;
+
+    @JsonProperty
+    private String resumed;
+
+    /*
+     * Non-standard property supplying a client with the current contest time (HH:MM:SS.uuu)
+     */
+    @JsonProperty
+    private String contest_time;
+
+    /**
+     * Fill in properties for contest state as per 2023-06 spec
+     *
+     * @param model place to get contest times from
+     * @param ci optional contest information.  If null, use the model's info.  This constructor
+     *      is also used by ContestInformation objects that may come from an event listener, so
+     *      we make it variable as it may not be from the current model.
+     */
+    public CLICSContestState(IInternalContest model, ContestInformation ci) {
+
+        if (model.getContestTime().isContestStarted()) {
+            if(ci == null) {
+                ci = model.getContestInformation();
+            }
+            ContestTime ct = model.getContestTime();
+
+            started = Utilities.getIso8601formatterWithMS().format(ct.getContestStartTime().getTime());
+            if (model.getContestTime().isPastEndOfContest()) {
+                Calendar endedDate = IJSONTool.calculateElapsedWalltime(model, ct.getContestLengthMS());
+                if (endedDate != null) {
+                    ended = Utilities.getIso8601formatterWithMS().format(endedDate.getTimeInMillis());
+                }
+            }
+
+            contest_time = ContestTime.formatTimeMS(ct.getElapsedMS());
+            if(ct.isContestRunning() == false) {
+                if(ended == null) {
+                    // This means the contest was paused - can't pause a contest if it is ended... so ended must be null
+                    paused = contest_time;
+                }
+            }
+            resumed = Utilities.getIso8601formatterWithMS().format(ct.getResumeTime().getTime());
+
+            String scoreboardFreezeDuration = ci.getFreezeTime();
+            Date thawedDate = null;
+
+            if (scoreboardFreezeDuration != null && scoreboardFreezeDuration.trim().length() > 0) {
+                long elapsed = ct.getElapsedSecs();
```

Comment:
Minor point, but I think it would be slightly clearer if this variable was renamed to `elapsedSecs` (like the subsequent variable which ends in "Secs").

Reply:
I removed the variable altogether and just called the method directly.  Only used once.  Next push.